### PR TITLE
Xarray and other upgrades

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #63 and #69.

- Stores data from load routine as an `xarray.Dataset`.  
- Saves version and commit (short hash) info in 'version.txt' file archived with output files after model is run
- Updates to documentation and setup procedures after #72. 
  - cleans out unused test_suite structure 
  - removes make command from travis
  - docstring updates
- Updates to unit tests
  - Expanded tests to cover optional outputs
  - regenerated test data with 2 time steps and neutral info
  - Imposed upper limits on pandas and xarray to fix travis server
- Minor fixes, including use of os.path.join

## Type of change

- Breaking change (data is now stored in a different format)
- This change requires a documentation update

## Context

This PR puts basic changes into play until f2py / netCDF is fully incorporated.  Version info will eventually be stored in netCDF metadata.

Upper level limits on pandas and xarray ***not*** a permanent solution.  Required for Travis CI to properly run python 2.7 tests.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by scivision** [comment]
FYI,
I have turned away from using f2py in situations where more than one instance of the Fortran module (in this case, Sami2) needs to run at once. Common and module variables from one instance of the imported f2py Fortran code can modify those of another instance, leading to non-deterministic (incorrect) results when more than one instance of the Fortran module via f2py is used at once.

Maybe you all have run into and/or solved that issue. For me thus far, the solution for parallel/concurrent use was to not use f2py, and instead to pass I/O from Python to Fortran either directly via pipes (stdin/stdout), or if there was too much data for pipes then via tempfile.NamedTemporaryFile or similar (using RAM dist where possible).

File: setup.cfg [hunk 1]
```diff
@@ -34,4 +34,6 @@ zip_safe = False
 packages = find:
 install_requires =
   numpy
+  pandas<0.25
```
Comment (scivision) [inline setup.cfg]: Could we add a comment here within setup.cfg so we don't forget it's just for Travis-I was puzzled till I saw the PR notes.
```ini
# FIXME: upper limit on pandas and xarray are just so Travis CI passes python 2.7 tests.
```

Comment (jklenzing) [reply setup.cfg]: Good idea.  Adding.

File: setup.py [hunk 2]
```diff
@@ -21,10 +21,10 @@
 
 if not path.isfile(path.join(fortran_path, 'sami2py.x')):
     try:  # py27 does not have shutil.which()
-        cmd = ['gfortran', '-fno-range-check', '-fno-automatic', '-ffixed-line-length-none',
-               '-o', 'sami2py.x']
-        src = ['nrlmsise00_modified.f', 'grid-1.00.f', 'sami2py-1.00.f', 'hwm93.f', 'hwm07e_modified.f90',
-               'apexcord.f90', 'hwm14.f90']
+        cmd = ['gfortran', '-fno-range-check', '-fno-automatic',
```
Comment (scivision) [inline setup.py]: This I think is fine for now. A more general approach for the future to consider is to have a `sami2py.build()` function that is automatically run on `import sami2py` if the Fortran module isn't built yet. I use that technique on my packages. This allows me to not depend on `setuptools()` and to switch compilers and options seamlessly/automatically. 

But I think for now, probably most people have Gfortran and setuptools--just noting there's a more general way forward if it becomes an issue.

**Pull request merged**